### ISSUE TITLE
Unhandled Exception: type 'Null' is not a subtype of type 'Map<dynamic, dynamic>' in type cast

### ISSUE THREAD

**Issue by Bilenne-Ndifor** [aws-amplify/amplify-flutter#5120]
### Description

i am new to AWS and i have built an application using flutter which stores information on datastore. it was working fine until i tried to make it sync online. since that i the configureamplify no longer works . keep getting the " Unhandled Exception: type 'Null' is not a subtype of type 'Map<dynamic, dynamic>' in type cast" error whenever i want to use data store
![Screenshot 2024-07-08 014820](https://github.com/aws-amplify/amplify-flutter/assets/80074211/e4ca8b59-c014-48dc-9871-15dda8949a4e)
thank you

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Steps to Reproduce

![Screenshot 2024-07-08 014652](https://github.com/aws-amplify/amplify-flutter/assets/80074211/2bfb604e-252c-4730-8f46-5f0ddc063463)

### Platforms

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

 3.22.2

### Amplify Flutter Version

2.1.0

### Deployment Method

Amplify CLI

### Schema

```GraphQL
# This "input" configures a global authorization rule to enable public access to
# all models in this schema. Learn more about authorization rules here: https://docs.amplify.aws/cli/graphql/authorization-rules
#input AMPLIFY { globalAuthRule: AuthRule = { allow: public } } # FOR TESTING ONLY!

type Completed @model @auth(rules: [{ allow: private }]){
  id: ID!
  sendername: String!
  senderemail: String!
  senderphone:String!
  recivername: String!
  reciveremail: String!
  reciverphone: String!
  reciverbankname: String!
  reciveraccount: String!
  transferpurpose: String!
  transfercurrency: String!
  transferammount: String!
  finaltransferammount: String!
  timerecieved: String!
  refcode: String
  phonenum:String!
  status: TransactionStatus!
  
}

type Inform @model @auth(rules: [{ allow: private }]){
  id: ID!
  search: String!
  rate: String!
  }

  type Wallet @model @auth(rules: [{ allow: private }]){
  refcode: String!
  amount: String!
  phone:String!
  }

enum TransactionStatus {
  UNKNOWN
  APPROVED
  REJECTED
  FINALIZED
  ERROR
}
```

**Comment by khatruong2009** [comment]
Hi @Bilenne-Ndifor, thank you for submitting this issue. We will take a look at it and get back to you with any updates.

**Comment by khatruong2009** [comment]
Hi @Bilenne-Ndifor, have you tried running `Amplify.Datastore.clear()` to possibly clear some old data from before the sync that might be causing the issue? Also, have you run `amplify push` in the terminal to make sure the resources have been pushed to AWS?

**Comment by Bilenne-Ndifor** [comment]
Yes i have cleared all records in my datastore . Also pushed all changes. 
Just to test things out , I also deleted the API recreated it and it still isn't working. 

After investigation the error pops up when ever i try to configure datastore or i add datastore under Amplift.addPlugin()

**Comment by khatruong2009** [comment]
Hi @Bilenne-Ndifor, I was able to reproduce the issue and am working to figure out the root cause. I'll get back to you with any updates.

**Comment by Bilenne-Ndifor** [comment]
Ok thank you.

**Comment by khatruong2009** [comment]
Hi @Bilenne-Ndifor, the fix for this has been released. Please upgrade your `amplify-datastore` version to `2.2.1`. I'm going to close this issue now, but please feel free to let us know if you have another issue that pops up or if this issue resurfaces.

**Comment by Bilenne-Ndifor** [comment]
Yes it works after flutter clear , but every time you restart The application you get the same problem .
It only starts working again after a flutter clean . I'm attempting to stop configuring the datastore once it is done the first time . I suspect thats the error in my code causing it . 
Thanks for the response

**Comment by Williamjr23** [comment]
I have the same issue, it only occurs in the Android version, every time I restart the app it gives me the same error: [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type 'Null' is not a subtype of type 'Map<dynamic, dynamic>' in type cast

**Comment by Bilenne-Ndifor** [comment]
This is a temporal solution. 
Update to datastore 2.2.1 
Run flutter clean 
Run pub get 
 Flutter run . 

It will run the first time(or atleast in my case) might give the same error after a few restarts .  
When i had the same error again ,i just flutter clean and it starts working again.
Still waiting for a lasting solution

**Comment by timonchev** [comment]
same in 2.4.2...

**Comment by harshitGit-01** [comment]
I am also experiencing this issue with version 2.5.0. Is there a temporary workaround available while you work on a fix?
